core-processor/src/executor.rs

```diff
@@ -511,18 +506,21 @@ where
         value_counter: ValueCounter::new(Default::default()),
         allocations_context: AllocationsContext::new(allocations, static_pages, 512.into()),
         message_context: MessageContext::new(
-            IncomingMessage::new(
-                Default::default(),
-                Default::default(),
-                payload
-                    .try_into()
-                    .map_err(|e| format!("Failed to create payload: {e:?}"))?,
-                gas_limit,
-                Default::default(),
-                Default::default(),
+            IncomingDispatch::new(
+                DispatchKind::Handle,
```

Comment:
how is this not `DispatchKind::Reply` ?

Reply:
execute_for_reply fn used to read state of wasms
it doesn't execute reply, but executes custom entry to collect error or only reply sent
so we need to throw here any kind that will allow replying (Init or Handle)